llvm/lib/CAS/MappedFileRegionBumpPtr.cpp

```diff
@@ -107,9 +112,15 @@ struct FileSizeInfo {
 } // end anonymous namespace
 
 Expected<MappedFileRegionBumpPtr> MappedFileRegionBumpPtr::create(
-    const Twine &Path, uint64_t Capacity, int64_t BumpPtrOffset,
+    const Twine &Path, uint64_t Capacity, uint64_t HeaderOffset,
     std::shared_ptr<ondisk::OnDiskCASLogger> Logger,
     function_ref<Error(MappedFileRegionBumpPtr &)> NewFileConstructor) {
+  uint64_t MinCapacity = HeaderOffset + sizeof(Header);
+  if (Capacity < MinCapacity)
+    return createStringError(
+        std::make_error_code(std::errc::no_space_on_device),
```

Comment:
This seems more like `invalid_argument` since it has no relation to the filesystem space.